### PR TITLE
Extend the Plan API to return plan templates and actual plans

### PR DESCRIPTION
**Pull request by rehammuzzamil**
Issue # 373

### CODE REVIEW

File: src/main/java/org/opensrp/web/rest/ClientFormResource.java [hunk 1]
```diff
@@ -75,6 +75,10 @@ public void setObjectMapper(ObjectMapper objectMapper) {
             boolean isDraft = Boolean.parseBoolean(isDraftParam.toLowerCase());
             clientFormMetadataList = clientFormService.getDraftsClientFormMetadata(isDraft);
         }
+//        else {
+//            boolean isDraft = Boolean.parseBoolean(isDraftParam.toLowerCase());
+//            clientFormMetadataList = clientFormService.getClientFormMetadata(isDraft);
+//        }
```
Comment (githengi) [inline src/main/java/org/opensrp/web/rest/ClientFormResource.java]: Remove these commennts

File: src/main/java/org/opensrp/web/rest/PlanResource.java [hunk 2]
```diff
@@ -84,21 +84,22 @@ public void setLocationService(PhysicalLocationService locationService) {
 	
 	@RequestMapping(value = "/{identifier}", method = RequestMethod.GET, produces = { MediaType.APPLICATION_JSON_VALUE })
 	public ResponseEntity<String> getPlanByUniqueId(@PathVariable("identifier") String identifier,
-	        @RequestParam(value = FIELDS, required = false) List<String> fields) {
+	        @RequestParam(value = FIELDS, required = false) List<String> fields , @RequestParam(value = "is_template", required = false) boolean isTemplateParam) {
```
Comment (githengi) [inline src/main/java/org/opensrp/web/rest/PlanResource.java:87]: define  "is_template" as a constant

File: src/main/java/org/opensrp/web/rest/ClientFormResource.java (hunk 1 above)
Comment (rehammuzzamil) [reply src/main/java/org/opensrp/web/rest/ClientFormResource.java]: Removed

File: src/main/java/org/opensrp/web/rest/PlanResource.java (hunk 2 above)
Comment (rehammuzzamil) [reply src/main/java/org/opensrp/web/rest/PlanResource.java]: Done

**Pull request merged**